handsontable/src/plugins/copyPaste/clipboardData/clipboardData.js

```diff
@@ -0,0 +1,546 @@
+   *
+   * Note: Used indexes refers to processed data, not to the instance of Handsontable. Please keep in mind that headers
+   * are handled separately from cells and they are recognised using negative indexes.
+   *
+   * @param {number} row Row index of cell which should be changed.
+   * @param {number} column Column index of cell which should be changed.
+   * @param {string} value Value for particular indexes.
+   */
+  setCellAt(row, column, value) {
+    if (this.isSerializedTable() === false) {
+      return;
+    }
+
+    const config = this.getMetaInfo();
+    const { data, nestedHeaders, colHeaders } = config;
+
+    if (row < 0) {
+      if (Array.isArray(nestedHeaders)) {
+        const rowRelative = row + nestedHeaders.length;
+
+        if (Array.isArray(nestedHeaders[rowRelative]) && isDefined(nestedHeaders[rowRelative][column])) {
+          nestedHeaders[rowRelative][column] = value;
+        }
+
+      } else if (Array.isArray(colHeaders)) {
+        if (isDefined(colHeaders[column])) {
+          colHeaders[column] = value;
+        }
+      }
+    } else if (row >= 0 && Array.isArray(data) && Array.isArray(data[row]) && isDefined(data[row][column])) {
+      data[row][column] = value;
+    }
+
+    this.overwriteInfo(config);
+  }
+
+  /**
+   * Gets header or cell values from the serialized dataset.
+   *
+   * Note: Used indexes refers to processed data, not to the instance of Handsontable. Please keep in mind that headers
+   * are handled separately from cells and they are recognised using negative indexes.
+   *
+   * @param {number} row Row index of cell which should be get.
+   * @param {number} column Column index of cell which should be get.
+   * @returns {undefined|string}
+   */
+  getCellAt(row, column) {
+    if (this.isSerializedTable() === false) {
+      return;
+    }
+
+    const config = this.getMetaInfo();
+    const { data, nestedHeaders, colHeaders } = config;
+
+    if (row < 0) {
+      if (Array.isArray(nestedHeaders)) {
+        const rowRelative = row + nestedHeaders.length;
+
+        if (Array.isArray(nestedHeaders[rowRelative]) && isDefined(nestedHeaders[rowRelative][column])) {
+          return nestedHeaders[rowRelative][column];
+        }
+
+      } else if (Array.isArray(colHeaders)) {
+        if (isDefined(colHeaders[column])) {
+          return colHeaders[column];
+        }
+      }
+    } else if (row >= 0 && Array.isArray(data) && Array.isArray(data[row]) && isDefined(data[row][column])) {
+      return data[row][column];
+    }
+  }
+
+  /**
+   * Checks whether serialized data is an array.
+   *
+   * @private
+   * @returns {boolean}
+   */
+  isSerializedTable() {
+    return true;
+  }
+
+  /**
+   * Checks whether serialized data is a Handsontable.
+   *
+   * @private
+   * @returns {boolean}
+   */
+  isSerializedHandsontable() {
+    return true;
+  }
+
+  /**
+   * Gets source of the copied data.
+   *
+   * @returns {string}
+   */
+  getSource() {
```

Comment:
After playing with the code for a while, I think the method name is misleading. It returns 3 values:
 * `handsontable` when we know that the pasted values are coming from the Handosntable;
 * `table` we know that the clipboard contains tabular data so we can parse it and process it;
 * `string` does not match any of the above.

The above values rather told me about the clipboard type, not the source.